File: src/main/java/com/mike/inject_model/exceptions/NotFoundException.java
package com.mike.inject_model.exceptions;

import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ResponseStatus;

/**
 * Exception
 */
@ResponseStatus(HttpStatus.NOT_FOUND)
public class NotFoundException extends ExceptionContract {

    public NotFoundException(String reason) {
        super(reason);
    }
}

File: src/main/java/io/github/robertomike/inject_model/exceptions/NotFoundException.kt
package io.github.robertomike.inject_model.exceptions

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ResponseStatus

/**
 * Exception
 */
@ResponseStatus(HttpStatus.NOT_FOUND)
class NotFoundException @JvmOverloads constructor(message: String, e: Throwable? = null) : ExceptionContract(message, e)